levels.go

```diff
@@ -574,7 +574,11 @@ func (s *levelsController) compactBuildTables(
 					return
 				}
 
-				if _, err := fd.Write(builder.Finish()); err != nil {
+				data, err := builder.Finish()
+				if err != nil {
+					resultCh <- newTableResult{nil, y.Wrapf(err, "failed to build table")}
```

Comment:
should we return from here?

Reply:
Nice catch. Thanks. Fixed in https://github.com/dgraph-io/badger/pull/816/commits/12fc3f5510e7bc15b512ca1f4eeb185de6697777